reana_commons/snakemake.py

```diff
@@ -36,15 +37,29 @@ def snakemake_validate(
     :param workdir: Path to working directory.
     :type workdir: string or None
     """
-    valid = snakemake(
-        snakefile=workflow_file,
-        configfiles=configfiles,
-        workdir=workdir,
-        dryrun=True,
-        quiet=True,
-    )
-    if not valid:
-        raise REANAValidationError("Snakemake specification is invalid.")
+    with SnakemakeApi(
+        OutputSettings(
+            quiet=True,
+        )
+    ) as snakemake_api:
+        try:
+            workflow_api = snakemake_api.workflow(
+                resource_settings=ResourceSettings(nodes=300),
```

Comment:
For the nodes settings, we already have `SNAKEMAKE_MAX_PARALLEL_JOBS` environmental variable that could be used instead of the hard-coded value of 300, even though this might "fully' work only when the validation would be moved to the server-side.

Still, you could preventively introduce in `reana_commons/config.py` something like:

```python
SNAKEMAKE_MAX_PARALLEL_JOBS = int(os.getenv("SNAKEMAKE_MAX_PARALLEL_JOBS", "300"))
"""Snakemake maximum number of jobs that can run in parallel."""
```

which might be better that hard-coding the value of 300.

(Similarly as we do in `reana_workflow_engine_snakemake/config.py`.)